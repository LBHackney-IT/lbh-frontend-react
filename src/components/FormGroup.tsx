import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import "lbh-frontend/lbh/objects/_form-group.scss";

/**
 * The proptypes for {@link FormGroup}.
 */
export interface FormGroupProps {
  id?: string;
  className?: string;
  /**
   * Whether there is an error in the form group or not.
   */
  error?: boolean;
  /**
   * The form elements.
   */
  children: React.ReactNode;
}

/**
 * A component used to group labels, controls, optional help text, and
 * form validation messaging. It can be used to wrap a {@link Fieldset}, or to
 * wrap nearly any other set of form elements.
 */
export const FormGroup: React.FunctionComponent<FormGroupProps> = (
  props: FormGroupProps
): JSX.Element => {
  const { id, className, error, children } = props;
  return (
    <div
      id={id}
      className={classNames("govuk-form-group lbh-form-group", className, {
        "govuk-form-group--error lbh-form-group--error": error
      })}
    >
      {children}
    </div>
  );
};

FormGroup.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  error: PropTypes.bool,
  children: PropTypes.node.isRequired
};
