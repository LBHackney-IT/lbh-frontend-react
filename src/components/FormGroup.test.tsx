import React from "react";
import { create } from "react-test-renderer";

import { FormGroup } from "./FormGroup";

it("renders correctly with all props", () => {
  const component = create(
    <FormGroup id="Testid" className="class1 class2" error={true}>
      TestFormGroupChildren
    </FormGroup>
  );

  expect(component).toMatchSnapshot();
});

it("renders correctly without optional props", () => {
  const component = create(<FormGroup>TestFormGroupChildren</FormGroup>);

  expect(component).toMatchSnapshot();
});
