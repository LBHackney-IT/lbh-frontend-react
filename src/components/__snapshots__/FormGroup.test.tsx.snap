// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="govuk-form-group lbh-form-group class1 class2 govuk-form-group--error lbh-form-group--error"
  id="Testid"
>
  TestFormGroupChildren
</div>
`;

exports[`renders correctly without optional props 1`] = `
<div
  className="govuk-form-group lbh-form-group"
>
  TestFormGroupChildren
</div>
`;
